lib/text-class/test/unit/Data/Text/ClassSpec.hs

```diff
@@ -159,14 +157,8 @@ instance Arbitrary Digits where
         pure (sign ++ str)
 
 instance Arbitrary Natural where
-    shrink x =
-        [ fromIntegral x
-        | x' <- shrink (fromIntegral x)
-        , x' >= 0
-        , x' /= x
-        ]
-    arbitrary =
-        fromIntegral . abs <$> arbitrary @Integer
+    shrink = shrinkIntegral
```

Comment:
Does this work without arithmetic underflow exceptions? I notice in the definition of `shrinkIntegral x` there is `-x`.

Reply:
Yes, I double checked the implementation of [shrinkIntegral](http://hackage.haskell.org/package/QuickCheck-2.13.2/docs/src/Test.QuickCheck.Arbitrary.html#shrinkIntegral) carefully before doing this change. The `-x` is negating those `x`-es which are negative in the first place -> makes them positive. But when shrinking Naturals there won't be any negative x-es in the first place, so that code path won't even be taken. The additional confirmation that I'm doing the right thing was the same implementation of `Arbitrary Natural` is used in [quickcheck-instances](http://hackage.haskell.org/package/quickcheck-instances-0.3.22/docs/src/Test.QuickCheck.Instances.Natural.html#line-19)